coq/coq-system.el

```diff
@@ -443,65 +466,76 @@ options of a few coq-project files does the right thing."
+LOADPATH, CURRENT-DIRECTORY, PRE-V85: see `coq-include-options'."
+;; coqtop always adds the current directory to the LoadPath, so don't
+  ;; include it in the -Q options.
+  (append (coq--clean-prog-args coq-prog-args)
+          (let* ((coq-load-path-include-current nil)
+                 (cmd (coq-include-prog-args loadpath current-directory pre-v85)))
+            cmd)))
+
+(defun coq-coqdep-prog-args (loadpath &optional current-directory pre-v85)
+  "Build a list of option for coqdep."
+  (append
+   (if (coq--post-v9) (list "dep") nil)
+   (coq-include-prog-args loadpath current-directory pre-v85)))
+
 (defun coq-coqc-prog-args (loadpath &optional current-directory pre-v85)
   "Build a list of options for coqc.
 LOADPATH, CURRENT-DIRECTORY, PRE-V85: see `coq-include-options'."
   ;; coqtop always adds the current directory to the LoadPath, so don't
   ;; include it in the -Q options.
-  (append (coq--clean-prog-args coq-prog-args)
-          (let ((coq-load-path-include-current nil)) ; Not needed in >=8.5beta3
-            (coq-coqdep-prog-args loadpath current-directory pre-v85))))
+  (append (if (coq--post-v9) (list "c") nil)
+          (coq-common-prog-args loadpath current-directory pre-v85)))
 
-;;XXXXXXXXXXXXXX
 ;; coq-coqtop-prog-args is the user-set list of arguments to pass to
 ;; Coq process, see 'defpacustom prog-args' in pg-custom.el for
 ;; documentation.
 
-;; Regular files should be of the form "valid_modulename.v" coq
-;; accepts lots of things as letter, this is probably much stricter.
-;; In practice it should be OK though, except maybe for non latin
-;; characters.
-(defun coq-regular-filename-p (s)
-  (let ((noext (file-name-base s)))
-    (string-match-p "\\`[[:alpha:]_][[:alnum:]_]*\\'" noext)))
-
 (defun coq-coqtop-prog-args (loadpath &optional current-directory pre-v85)
   ;; coqtop always adds the current directory to the LoadPath, so don't
   ;; include it in the -Q options. This is not true for coqdep.
   "Build a list of options for coqc.
 LOADPATH, CURRENT-DIRECTORY, PRE-V85: see `coq-coqc-prog-args'."
-  (let ((topfile-supported (coq--supports-topfile)))
-    (append
-     ;; it is better to inform coqtop of the name of the current module
-     ;; using the -topfile option, so that coqtop understands references
-     ;; to it. But don't insist if this would fail (because of wrong
-     ;; file name): Some people want to script .v files without ever
-     ;; compiling them.
-     (if (and topfile-supported buffer-file-name
-              (coq-regular-filename-p buffer-file-name))
-         (cons "-topfile" (cons buffer-file-name nil))
-       (if (and topfile-supported buffer-file-name)
-           (message "Warning: this Coq buffer is probably not compilable \
-because of its name, no -topfile option set."))
-       nil)
-     (cons "-emacs" (coq-coqc-prog-args loadpath current-directory pre-v85)))))
+  (append
+   (if (coq--post-v9) (list "top") nil)
```

Comment:
This seems wrong. We can look at the exe name (as done at line 131) or maybe the output of calling the exe with `-help`.